packages/mwp-auth-plugin/src/index.js

```diff
@@ -1,342 +1,57 @@
-				// programmer error or catastrophic auth failure - throw exception
-				throw new ReferenceError(
-					'No grant token provided - cannot obtain access token'
-				);
-			}
-
-			accessUrl.searchParams.append('grant_type', grant_type);
-			if (grant_type === 'anonymous_code') {
-				accessUrl.searchParams.append('code', token);
-			}
-			if (grant_type === 'refresh_token') {
-				accessUrl.searchParams.append('refresh_token', token);
-			}
-
-			logger.info(
-				{
-					type: 'request',
-					direction: 'out',
-					info: {
-						url: serverAppConfig.oauth.access_url,
-						method: 'get',
-					},
-				},
-				`Outgoing request GET ${serverAppConfig.oauth.access_url}?${grant_type}`
-			);
-
-			const startTime = new Date();
-			return Observable.fromPromise(fetch(accessUrl.toString(), requestOpts))
-				.timeout(serverAppConfig.api.timeout)
-				.do(() => {
-					logger.info(
-						{
-							type: 'response',
-							direction: 'in',
-							info: {
-								url: serverAppConfig.oauth.access_url,
-								method: 'get',
-								elapsedTime: new Date() - startTime,
-							},
-						},
-						`Incoming response GET ${serverAppConfig.oauth
-							.access_url}?${grant_type}`
-					);
-				})
-				.mergeMap(tryJSON(serverAppConfig.oauth.access_url));
-		};
-	};
-};
-
-const refreshToken$ = refresh_token =>
-	Observable.of({
-		grant_type: 'refresh_token',
-		token: refresh_token,
-	});
-
-/**
- * Curry a function that will get a new auth token for a passed-in request.
- * For an anonymous auth, the request header information is used to determine
- * the location and language of the anonymous member
- *
- * @param {Object} serverAppConfig the Hapi `server.settings.app` config object
- */
-export const getRequestAuthorizer$ = serverAppConfig => {
-	const redirect_uri = 'http://www.meetup.com/'; // required param set in oauth consumer config
-	const anonymousCode$ = getAnonymousCode$(serverAppConfig, redirect_uri);
-	const accessToken$ = getAccessToken$(serverAppConfig, redirect_uri);
-
-	// if the request has a refresh_token, use it. Otherwise, get a new anonymous access token
-	return ({ headers, state: { refresh_token }, server: { app: { logger } } }) =>
-		Observable.if(
-			() => refresh_token,
-			refreshToken$(refresh_token),
-			anonymousCode$
-		)
-			.mergeMap(accessToken$(headers))
-			.do(verifyAuth(logger));
-};
-
-export const getAuthenticate = authorizeRequest$ => (request, reply) => {
-	const { server: { app: { logger } }, raw: { req } } = request;
-	logger.debug({ req }, 'Authenticating request');
-	return authorizeRequest$(request)
-		.do(request => {
-			logger.debug({ req }, 'Request authenticated');
-		})
-		.subscribe(
-			request => {
-				const credentials =
-					request.state[getAuthType(request)] ||
-					request.plugins.requestAuth.oauth_token;
-				reply.continue({ credentials, artifacts: credentials });
-			},
-			err => reply(err, null, { credentials: null })
-		);
+const mockCookies = {
+	MEETUP_MEMBER:
```

Comment:
waaaaa :) but really we should describe this :)

Reply:
It's described here https://github.com/meetup/meetup-web-platform/pull/412/files#diff-174f25f23863d6b2b6ba3aa79cfcc151R19

But I can add another comment here